### ISSUE TITLE
Limit the listed dependencies to those that are actually reachable

### ISSUE THREAD

**Issue by bruceg** [DataDog/rust-license-tool#126]
The data source underlying `rust-license-tool` lists all entries in the dependency tree, even if they can't be reached under any actual combination of top-level feature flags. While we can't reasonably expand every possible combination of the top-level feature flags (Vector has 228 which leads to about 10^440 unique combinations), we should be able to reduce that dependency tree by producing sets of possible feature flags for each crate and thereby eliminate sub-dependencies that can't be turned on. This would reduce the size of the generated licenses file, producing a more accurate reflection of what components are actually used.

**Comment by jszwedko** [comment]
Is https://github.com/rust-lang/cargo/issues/10801 related to this?

**Comment by bruceg** [comment]
Yes, that's the one. I was looking for that link when I wrote the above but couldn't find it. There may be opportunities for us to solve this without support from that issue, however.
